File: 2899-last-visited-integers/2899-last-visited-integers.cpp
class Solution {
public:
    vector<int> lastVisitedIntegers(vector<string>& words) {
        vector<int>v;
        vector<int>temp;
        int cnt =0;
        for(int i=0;i<words.size();++i){
            if(words[i]=="prev"){
                cnt+=1;
                if(temp.size()>=cnt){
                    v.push_back(temp[temp.size()-cnt]);
                }else{
                    v.push_back(-1);
                }
            }else{
                cnt=0;
                temp.push_back(stoi(words[i]));
            }
        }
        return v;
    }
};

File: 2899-last-visited-integers/2899-last-visited-integers.py
class Solution:
    def lastVisitedIntegers(self, words: List[str]) -> List[int]:
        temp, ans = [], []
        cnt = 0
        for i in range(len(words)):
            if words[i] == "prev":
                cnt += 1
                if len(temp)>=cnt:
                    ans.append(temp[len(temp)-cnt])
                else:
                    ans.append(-1)
            else:
                cnt = 0
                temp.append(int(words[i]))
        return ans
                